### PR TITLE
UCSBSubjects `GET` (index) and `POST` (create)

### PR DESCRIPTION
**Pull request by SwiftWinds**
# Overview

This pr creates two endpoints: `GET /api/UCSBSubjects/all` and `POST `/api/UCSBSubjects/post`. The first endpoint returns a list of all UCSB subjects while the latter allows the caller to create a new UCSB subject and store it in the database.

# Issues Addressed

#18 

## Reminders (delete this section before merging or as code is merged)
- [x] Explicitly link to this pull request each issue that it closes
- [x] Assign one and ideally two reviewers to review this pull request. 
- [x] Reviewers should not just approve with "Looks good!" Instead, reviewers should carefully go through the code and use GitHub's reviewer interface to ask questions, make comments, and suggest changes.

### CODE REVIEW

File: src/main/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectController.java [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+@Api(description = "UCSBSubjects")
+@RequestMapping("/api/UCSBSubjects/")
+@RestController
+@Slf4j
+public class UCSBSubjectController extends ApiController {
+
+//    /**
```
Comment (tekaireb) [inline src/main/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectController.java]: Perhaps we could delete the Todo code?

File: src/main/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectController.java [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    ObjectMapper mapper;
+
+    @ApiOperation(value = "List all UCSB Subjects")
+    @GetMapping("/all")
+    public Iterable<UCSBSubject> allUCSBSubjects() {
+        loggingService.logMethod();
+        return ucsbSubjectRepository.findAll();
+    }
+
+//    @ApiOperation(value = "Get a single todo (if it belongs to current user)")
```
Comment (tekaireb) [inline src/main/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectController.java]: Unnecessary Todo code

File: src/main/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectController.java [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+        ucsbSubject.setSubjectTranslation(subjectTranslation);
+        ucsbSubject.setDeptCode(deptCode);
+        ucsbSubject.setCollegeCode(collegeCode);
+        ucsbSubject.setRelatedDeptCode(relatedDeptCode);
+        ucsbSubject.setInactive(inactive);
+
+        return ucsbSubjectRepository.save(ucsbSubject);
+    }
+
+//    @ApiOperation(value = "Delete a Todo owned by this user")
```
Comment (tekaireb) [inline src/main/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectController.java]: Todo code

**Review by tekaireb — CHANGES_REQUESTED** [review]
Looks good! There's just a lot of leftover (commented out) `Todo` code that could be removed. Otherwise, great job!

File: src/main/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectController.java (hunk 1 above)
Comment (nguglielmin) [reply src/main/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectController.java]: I think for the `main` branch it is probably smart to delete the `Todo` code.

**Pull request merged**